### DISCUSSION TITLE
What type should children be for DropdownMenu component to accept it?

### DISCUSSION THREAD

**Discussion by aprilmintacpineda** [heroui-inc/heroui#3462] [Help]
I'm trying to do this:

```tsx
import Burger from '@/app/icons/burger';
import {
  Button,
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownTrigger,
} from '@nextui-org/react';
import { type PropsWithChildren } from 'react';

export const QuickActionMenuItem = DropdownItem;

export default function QuickActionMenu({
  children,
}: PropsWithChildren) {
  return (
    <div className="fixed right-4 bottom-4">
      <Dropdown>
        <DropdownTrigger>
          <Button
            isIconOnly
            size="lg"
            className="rounded-full"
            variant="shadow"
            color="primary"
          >
            <Burger />
          </Button>
        </DropdownTrigger>
        <DropdownMenu>{children}</DropdownMenu>
      </Dropdown>
    </div>
  );
}
```

But I get:

```
Type 'ReactNode' is not assignable to type 'CollectionChildren<object>'.
  Type 'undefined' is not assignable to type 'CollectionChildren<object>'.
```

But that doesn't make sense because I could literally pass any children there.

```tsx
<DropdownMenu>
   <p>THIS WORKS</p>
</DropdownMenu>
```

This is one of those components that are pretty hard to extend.

**Comment by yankee0** [comment]
I have the same problem, did you solve it?

<img width="660" alt="Screenshot 2025-01-05 at 20 13 01" src="https://github.com/user-attachments/assets/94908432-276c-4d15-ac67-47dc178c1ae1" />
<img width="688" alt="Screenshot 2025-01-05 at 20 13 33" src="https://github.com/user-attachments/assets/a0f952ba-a24d-4e5d-a630-c7b8b10515de" />
<img width="754" alt="Screenshot 2025-01-05 at 20 14 13" src="https://github.com/user-attachments/assets/cb288d5e-31ce-4da6-9a52-658fb5a2ff32" />
